### DISCUSSION TITLE
Existe algum limite de retorno por requisição ?

### DISCUSSION THREAD

**Discussion by WesleyjanLacerda** [lucashpmelo/node-mde#18] [General]
Olá, minha dúvida é a seguinte, toda vez que faço uma requisição "distribuicao.consultaUltNSU('000000000000050')", tenho que ficar passando de 50 em 50 os NSU, e se eu passar em menos de 1h, digamos "000000000000050" e depois tento "000000000000100" recebo um bloqueio de 1h, é assim mesmo ? ou tem algum outro critério ?

**Comment by lucashpmelo** [comment]
Boa noite,

Você sempre deve observar três valores retornados, `cStat`, `ultNSU` e `maxNSU`.

Caso o valor do `cStat` seja `"137"` ou `"656"` a consulta deve ser interrompida por 1 hora.
- O `137` indica que a consulta chegou no ultimo NSU gerado pela receita e não existem mais documentos para serem retornados no momento.
- Já o `656` indica que a consulta foi feita na ordem errada, ou o tempo de bloqueio de 1 hora não foi respeitado.

Se o `cStat` for igual a `"138"` significa que foram encontrados novos documentos, nesse caso você deve comparar os valores de `ultNSU` e `maxNSU` retornados. O `ultNSU` é o valor que deve ser usado na próxima consulta, e o `maxNSU` é a quantidade máxima de NSU gerados até o momento.
- Caso o `ultNSU` retornado seja menor que o `maxNSU` você pode continuar a consulta passando o valor do `ultNSU`.
- Já se o `ultNSU` e `maxNSU` forem iguais você deve parar a consulta para evitar receber um `cStat: "137"`.
